src/VirtoCommerce.News.Web/App/src/modules/vc-news/pages/news-article-list.vue

```diff
@@ -0,0 +1,213 @@
+  expanded: true,
+  closable: true,
+});
+
+defineEmits<Emits>();
+
+const { showConfirmation } = usePopup();
+
+const { t } = useI18n({ useScope: "global" });
+const { openBlade } = useBladeNavigation();
+
+const { newsArticles, newsArticlesCount, pagesCount, pageIndex, searchNewsArticles, searchQuery, loadingNewsArticles, deleteNewsArticles } = useNewsArticleList();
+
+const searchKeyword = ref();
+const selectedItemId = ref<string>();
+const selectedIds = ref<string[]>([]);
+
+watch(
+  () => props.param,
+  (newVal) => {
+    selectedItemId.value = newVal;
+  },
+  { immediate: true },
+);
+
+const bladeToolbar = computed((): IBladeToolbar[] => [
+  {
+    id: "refresh",
+    title: computed(() => t("VC_NEWS.PAGES.LIST.TOOLBAR.REFRESH")),
+    icon: "material-refresh",
+    async clickHandler() {
+      await reload();
+    },
+  },
+  {
+    id: "add",
+    title: computed(() => t("VC_NEWS.PAGES.LIST.TOOLBAR.ADD")),
+    icon: "material-add",
+    clickHandler: async () => {
+      openDetailsBlade(undefined);
+    },
+  },
+  {
+    id: "delete",
+    title: computed(() => t("VC_NEWS.PAGES.LIST.TOOLBAR.DELETE")),
+    icon: "material-delete",
+    disabled: selectedIds.value.length === 0,
+    clickHandler: async () => {
+      if (
+        await showConfirmation(
+          t("VC_NEWS.PAGES.LIST.ALERTS.DELETE_SELECTED_CONFIRMATION_MESSAGE", { count: selectedIds.value.length }),
+        )
+      ) {
+        await deleteNewsArticles({ ids: selectedIds.value });
+        selectedIds.value = [];
+        await reload();
+      }
+    },
+  }
+]);
+
+const columns = ref<ITableColumns[]>([
+  {
+    id: "name",
+    title: computed(() => t("VC_NEWS.PAGES.LIST.TABLE.HEADER.NAME")),
+    alwaysVisible: true,
+    sortable: true,
+    width: "50%"
+  },
+  {
+    id: "storeId",
+    title: computed(() => t("VC_NEWS.PAGES.LIST.TABLE.HEADER.STORE")),
+    alwaysVisible: true,
+    sortable: true,
+    width: "30%"
+  },
+  {
+    id: "isPublished",
+    title: computed(() => t("VC_NEWS.PAGES.LIST.TABLE.HEADER.IS_PUBLISHED")),
+    alwaysVisible: true,
+    sortable: true,
+    width: "20%"
+  },
+]);
+
+const title = computed(() => t("VC_NEWS.PAGES.LIST.TITLE"));
+
+const reload = async () => {
+  await searchNewsArticles();
+};
+
+const onItemClick = (item: { id: string }) => {
+  openDetailsBlade(item.id);
+};
+
+const openDetailsBlade = (id: string | undefined) => {
+  openBlade({
+    blade: markRaw(NewsArticleDetails),
+    param: id ?? undefined,
+    onOpen() {
+      selectedItemId.value = id ?? undefined;
+    },
+    onClose() {
+      selectedItemId.value = undefined;
+    },
+  });
+}
+
+const onSearchChange = (searchKeyword: string | undefined) => {
+  searchQuery.value.searchPhrase = searchKeyword;
+  searchNewsArticles();
+}
+
+const onPaginationClick = (page: number) => {
+  pageIndex.value = page;
+  searchNewsArticles();
+}
+
+const onHeaderClick = async (item: ITableColumns) => {
```

Comment:
Here, instead of complex custom sorting logic, you can also use the [useTableSort composable.](https://docs.virtocommerce.org/platform/developer-guide/custom-apps-development/vc-shell/Essentials/ui-components/vc-table/#handling-sorting-with-usetablesort)